### PR TITLE
Update packages to last versions - 8 for sdk 17.8 for vs

### PR DESCRIPTION
**Pull request by vzarytovskii**
This updates most important packages we use.

### CODE REVIEW

**Comment by psfinaki** [comment]
FWIW don't update Roslyn just yet (I don't know if you were going to) - it will break a code fix otherwise. I will update Roslyn manually next week.

**Comment by vzarytovskii** [comment]
> FWIW don't update Roslyn just yet (I don't know if you were going to) - it will break a code fix otherwise. I will update Roslyn manually next week.

Are you sure that's the version which already has changes? Because I'll have to rollback a bunch of other packages in this case.

**Comment by vzarytovskii** [comment]
@MichaelSimons could you please advise on how to see if the packages we need are in arcade/darc (besides MSBuild ones). Like BCL or StringTools, etc?

**Comment by MichaelSimons** [comment]
@vzarytovskii  - sorry for the delayed response.  For more responsiveness, it is best to mention @dotnet/source-build-internal.

Regarding the reported prebuilts.  

1. For the msbuild related versions, I noticed the corresponding entry in the Versions.Details.xml file was not updated.   Updating this will eliminate a number of the prebuilts.
2. For the remaining prebuilts that are 8.0 versions, dependencies should be declared in the Versions.Details.xml file with the source-build attribute set (it needs/can only be set on one dependency per repo.)
3. Regarding System.Runtime.CompilerServices.Unsafe, why is a dependency being made on a 7.0 preview version?  This smells.  Can this be made on the 7.0 G.A version?

**Comment by vzarytovskii** [comment]
> @vzarytovskii - sorry for the delayed response. For more responsiveness, it is best to mention @dotnet/source-build-internal.
> 
> Regarding the reported prebuilts.
> 
> 1. For the msbuild related versions, I noticed the corresponding entry in the Versions.Details.xml file was not updated.   Updating this will eliminate a number of the prebuilts.

Tanks, will update there as well, and add darc dependencies so it's updated by bot.

> 2. For the remaining prebuilts that are 8.0 versions, dependencies should be declared in the Versions.Details.xml file with the source-build attribute set (it needs/can only be set on one dependency per repo.)

Do you, by any chance, have an example of it?

> 3. Regarding System.Runtime.CompilerServices.Unsafe, why is a dependency being made on a 7.0 preview version?  This smells.  Can this be made on the 7.0 G.A version?

I don't think there's a 7.0 GA version, only previews for now.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -6,27 +6,39 @@
...
     </Dependency>
-    <Dependency Name="Microsoft.Build.Utilities.Core" Version="17.7.0-preview-23217-02">
+    <Dependency Name="Microsoft.Build.Utilities.Core" Version="17.8.0-preview-23411-01">
       <Uri>https://github.com/dotnet/msbuild</Uri>
       <Sha>2cbc8b6aef648cf21c6a68a0dab7fe09a614e475</Sha>
     </Dependency>
+    <Dependency Name="System.Collections.Immutable" Version="8.0.0-rc.2.23457.7">
+      <Uri>https://github.com/dotnet/runtime</Uri>
+      <Sha>da3500bb02343b1d0424c74ccdddbc592b5b3f4f</Sha>
+    </Dependency>
```
Comment (MichaelSimons) [inline eng/Version.Details.xml:33]: One (and only one) of the dependencies from runtime needs the following adornment

```
<SourceBuild RepoName="runtime" ManagedOnly="false" />
```

This should eliminate the remaining prebuilts.

Comment (vzarytovskii) [reply eng/Version.Details.xml:33]: Great, thanks a lot!

Comment (MichaelSimons) [reply eng/Version.Details.xml:34]: Looks like the source-build leg is green now. 🎉

**Comment by vzarytovskii** [comment]
Going to postpone it, since it causes mef loading issues in 17.8

**Pull request closed without merging**